### PR TITLE
[NCL-471]: Rest get build record log returns 204 when it is empty

### PR DESCRIPTION
**Pull request by jbartece**

### CODE REVIEW

File: rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        int logId = 1;
+        String logContent = "LOG CONTENT";
+        BuildRecordEndpoint buildRecordEndpoint = getLogsPrepareEndpoint(logId, logContent);
+
+        // then
+        assertEquals(200, buildRecordEndpoint.getLogs(logId).getStatus());
+    }
+
+    private BuildRecordEndpoint getLogsPrepareEndpoint(int logId, String logContent) {
+        // given
```
Comment (slaskawi) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: //given to be removed

File: rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        BuildRecordRepository buildRecordRepository = mock(BuildRecordRepository.class);
+        BuildRecordProvider buildRecordProvider = new BuildRecordProvider(buildRecordRepository, null);
+        BuildRecordEndpoint buildRecordEndpoint = new BuildRecordEndpoint(buildRecordProvider, null);
+        BuildRecord buildRecord = mock(BuildRecord.class);
+
+        // when
```
Comment (slaskawi) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: //when to be removed

File: rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * 
+ * @author Jakub Bartecek <jbartece@redhat.com>
+ *
+ */
+public class BuildRecordEndpointTest {
+
+    @Test
+    public void getLogsNoContentTest() {
+        int logId = 1;
```
Comment (slaskawi) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: Please add //given //when

File: rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        BuildRecordEndpoint buildRecordEndpoint = getLogsPrepareEndpoint(logId, "");
+
+        // then
+        assertEquals(204, buildRecordEndpoint.getLogs(logId).getStatus());
+    }
+
+    @Test
+    public void getLogsWithContentTest() {
+        int logId = 1;
```
Comment (slaskawi) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: The same here

File: rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import static org.mockito.Mockito.mock;
+
+import org.jboss.pnc.datastore.repositories.BuildRecordRepository;
+import org.jboss.pnc.model.BuildRecord;
+import org.jboss.pnc.rest.provider.BuildRecordProvider;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ *
```
Comment (slaskawi) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: Add a valid Javadoc or remove it.

File: rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java [hunk 6]
```diff
@@ -119,17 +121,25 @@ public BuildRecordRest getSpecific(Integer id) {
         }
         return null;
     }
+    
 
-    public StreamingOutput getLogsForBuildId(Integer id) {
+    public String getBuildRecordLog(Integer id) {
```
Comment (slaskawi) [inline rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java]: Maybe we should remove this method and preserve only the streaming one?

Comment (jbartece) [inline rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java]: It is needed to find out if the content is empty or not. If the content is not empty the StreamingOutput is created after that

File: rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java (hunk 1 above)
Comment (jbartece) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: Where should I add it, when I outlined it to the extra method?

File: rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java (hunk 6 above)
Comment (slaskawi) [inline rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java]: Ok, that makes sense.

File: rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java (hunk 1 above)
Comment (slaskawi) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: //given //when //then should be added only to the test method's body. This blog post describes it really well: http://monkeyisland.pl/2009/12/07/given-when-then-forever/

Comment (jbartece) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: I have two choices:
  * put it to the method body, but there will be duplication of code in two tests
  * or outline it

So, are you for the first option?

Comment (slaskawi) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: I would prefer if you remove it from here and implement the test like the following:
```
@Test
    public void getLogsWithContentTest() {
        //given
        int logId = 1;
        String logContent = "LOG CONTENT";

        //when
        BuildRecordEndpoint buildRecordEndpoint = getLogsPrepareEndpoint(logId, logContent);

        // then
        assertEquals(200, buildRecordEndpoint.getLogs(logId).getStatus());
    }
```

Comment (jbartece) [inline rest/src/test/java/org/jboss/pnc/rest/endpoint/BuildRecordEndpointTest.java]: OK

**Pull request merged**